integration/tests/cook/test_basic.py

```diff
@@ -551,6 +551,34 @@ def test_change_retries(self):
         job = util.wait_for_job(self.cook_url, job_uuid, 'completed')
         self.assertEqual('success', job['state'], 'Job details: %s' % (json.dumps(job, sort_keys=True)))
 
+    def test_change_failed_retries(self):
+        job_specs = util.minimal_jobs(2, max_retries=1, command='sleep 5')
+        try:
+            jobs, resp = util.submit_jobs(self.cook_url, job_specs)
+            self.assertEqual(resp.status_code, 201)
+            # wait for first job to start running, and kill it
+            job_uuid = jobs[0]
+            util.wait_for_job(self.cook_url, job_uuid, 'running')
+            util.kill_jobs(self.cook_url, [job_uuid])
+            job = util.load_job(self.cook_url, job_uuid)
+            self.assertEqual('failed', job['state'])
+            # retry both jobs, but with the failed_only=true flag
+            resp = util.retry_jobs(self.cook_url, retries=4, failed_only=True, jobs=jobs)
+            self.assertEqual(201, resp.status_code, resp.text)
+            job = util.load_job(self.cook_url, job_uuid)
+            self.assertIn(job['status'], ['waiting', 'running'])
+            jobs = util.wait_for_jobs(self.cook_url, jobs, 'completed')
+            # We expect both jobs to be completed successfully now.
+            # The first job (which we killed and retried) should have 2 retries remaining.
```

Comment:
I don't understand why the first job would have 2 retries instead of 3

Reply:
Each run counts against the retries-remaining count. Since it ran twice (run, kill, restart, complete), it's left with 2/4.